pkg/yatee/yatee.go

```diff
@@ -450,20 +454,26 @@ func recurse(input map[interface{}]interface{}, settings map[string]interface{},
 					if err != nil {
 						return nil, err
 					}
-					for valk, valv := range val {
-						if valk != "@else" {
-							res[valk] = valv
+					for _, vval := range val {
+						if vval.Key != "@else" {
+							res = append(res, yaml.MapItem{Key: vval.Key, Value: vval.Value})
 						}
 					}
 				} else {
-					elseClause, ok := mii["@else"]
-					if ok {
-						elseDict, ok := elseClause.(map[interface{}]interface{})
+					var elseClause interface{}
+					for _, miiv := range mii {
```

Comment:
nit: extract to a sub-function, say `findElseClause` ?
```go
if elseClause := findElseClause(mii); elseClause != nil{
...
```

Reply:
Unless i'm misunderstanding you, I think this whole function needs refactoring into multiple smaller functions, i'm not sure whether doing just 1 piece is worthwhile and I'd rather do it all at once to keep the code consistent. WDYT?

Reply:
Yep you're right, it's just that you added this piece of code, whereas the rest was already there.

Reply:
And by the way it was just nit picking